File: Practice/Greedy/q03/q03.py
s = input()
zero = 0
one = 0

if s[0] == "1":
    zero += 1
else:
    one += 1

for idx in range(len(s)-1):
    if s[idx] != s[idx+1]:
        if s[idx+1] == "1":
            zero += 1
        else:
            one += 1
print(min(zero, one))

File: Practice/Greedy/q03/q03.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int result, zero, one;
string s;

int main(void){
    cin >> s;
    if(s[0] == '1'){
        zero++;
    }else{
        one++;
    }
    for(int i = 1; i < s.length()-1; i++){
        if(s[i] != s[i+1]){
            if(s[i+1] == '1'){
                zero++;
            }else{
                one++;
            }
        }
    }
    printf("%d\n", min(zero, one));
}